### ISSUE TITLE
Colab error

### ISSUE THREAD

**Issue by nikky4D** [cwkx/GON#1]
Hi,

I would like to see the effect of the network on a single image. Is this a good use case, even though the work is targeted to a dataset? 

Also, In the colab file, at L_inner, indicated below, 
```

for step in range(501):
    # sample a batch of data
    x, t = next(train_iterator)
    x, t = x.to(device), t.to(device)
    x = x.permute(0, 2, 3, 1).reshape(batch_size, -1, n_channels)

# compute the gradients of the inner loss with respect to zeros (gradient origin)
    z = torch.zeros(batch_size, 1, num_latent).to(device).requires_grad_()
    z_rep = z.repeat(1,c.size(1),1)
    g = F(torch.cat((c, z_rep), dim=-1))
    **L_inner = ((g - x)**2).sum(1).mean()**
    z = -torch.autograd.grad(L_inner, [z], create_graph=True, retain_graph=True)[0]
```
what is t used for? it does not appear to be used in later parts of the file? or maybe I missed it. 

`

**Comment by cwkx** [comment]
hi nikky, t is the targets (labels) of the MNIST dataset. We don't use it (although we kept it there just in case anyone wants to try the classifier derivation we show in Eqn 2). The SIREN function F works very well for a single image, to modify this code back to just a simple SIREN, change num_latent=0,  then you can comment out the inner_loss bit, as that's not needed:

```   
z = torch.zeros(batch_size, 1, num_latent).to(device).requires_grad_()
z_rep = z.repeat(1, c.size(1), 1)
g = F(torch.cat((c, z_rep), dim=-1))
L_outer = ((g - x)**2).sum(1).mean()
optim.zero_grad()
L_outer.backward()
optim.step()
````

then visualise g reshaped back to the image size:
g = torch.clamp(g, 0, 1).permute(0,2,1).reshape(-1, n_channels, img_size, img_size)
